conda_index/index/__init__.py

```diff
@@ -822,15 +827,11 @@ def _write_repodata(self, subdir, repodata, json_filename):
         Write repodata to :json_filename, but only if changed.
         """
         repodata_json_path = join(self.channel_root, subdir, json_filename)
-        # add newline here (historic) so .bz2, .zst is identical to repodata.json
-        new_repodata = (
-            json.dumps(
-                repodata,
-                indent=2,
-                sort_keys=True,
-            )
-            + "\n"
-        )
+        if self.compact_json:
+            new_repodata = self.json_dumps(repodata)
+        else:
+            # add newline here (historic) so .bz2, .zst is identical to repodata.json
+            new_repodata = self.json_dumps(repodata) + "\n"
```

Comment:
```suggestion
        new_repodata = self.json_dumps(repodata)
        if self.compact_json:
            # add newline here (historic) so .bz2, .zst is identical to repodata.json
            new_repodata += "\n"
```

Maybe less repetitive like that? I wonder though if the newline logic belongs to `json_dumps`

Reply:
I concur, a `self.json_dumps(repodata, append_newline=True)` or something might do the trick.

So this could be toggled here or alternatively queried inline in the `json_dumps` method.

```suggestion
        # add newline here (historic) so .bz2, .zst is identical to repodata.json
        new_repodata = self.json_dumps(repodata, append_newline=self.compact_json)
```